# Inferential Analysis Notebook

## Overview
The inferential analysis in this notebook will explore statistically significant relationships within the Utah real estate dataset. By performing hypothesis tests, we aim to determine the impact of key property attributes, such as the number of bedrooms, lot size, and presence of a garage, on listing prices. This analysis will help identify which features hold genuine predictive power and are not just coincidentally related to price, setting the stage for more accurate predictive modeling in the following steps.

## Inferential Analysis Questions
- Does the number of bedrooms significantly affect the listing price?
- Is there a significant difference in listing prices for properties built before vs. after 2000?
- Do properties with larger lot sizes (lot_sqft) command higher listing prices?
- Does having a garage significantly influence the listing price?
- Is there a significant difference in prices across different property types?


### Hypothesis Tests

In [16]:
#Imports 
import pandas as pd
from scipy.stats import f_oneway, ttest_ind

In [17]:
#Loading the Clean & Transformed Data
path = ('../data/cleaned_real_estate_utah.csv')
cleanedrs_data = pd.read_csv(path)

### Does the number of bedrooms significantly affect the listing price?

Hypotheses:

- Null Hypothesis (H0): The average listing price does not significantly differ between properties with different numbers of bedrooms.
- Alternate Hypothesis (H1): The average listing price differs significantly between properties with different numbers of bedrooms.

**Test:** We use an ANOVA (Analysis of Variance) test since we are comparing the mean prices across multiple groups (different bedroom counts).


In [18]:
# Group prices by number of bedrooms
groups = [group["listPrice"].values for name, group in cleanedrs_data.groupby("beds")]

#Perform the ANOVA test to compare average prices across bedroom groups
stat, p_value = f_oneway(*groups)

#Results
print("ANOVA Test for Bedrooms and Listing Price:")
print("Statistic:", stat)
print("P-value:", p_value)

ANOVA Test for Bedrooms and Listing Price:
Statistic: 70.30960792986622
P-value: 6.754403814141851e-146


With a p-value of 6.754e-146, which is far below 0.05, we reject the null hypothesis. This result suggests that the number of bedrooms does significantly affect the listing price. In practical terms, properties with different numbers of bedrooms generally have different average prices, meaning buyers might pay more as bedroom count increases.

### Is There a Significant Difference in Listing Prices for Properties Built Before vs. After 2000?

Hypotheses:

- Null Hypothesis (H0): The average listing price for properties built before 2000 is not significantly different from properties built after 2000.
- Alternate Hypothesis (H1): The average listing price for properties built before 2000 is significantly different from properties built after 2000.

**Test:** Since we are comparing the mean prices between two independent groups (before and after 2000), a t-test is appropriate.

In [19]:
# Divide data into two groups: properties built before 2000 and properties built in or after 2000
before_2000 = cleanedrs_data[cleanedrs_data['year_built'] < 2000]['listPrice']
after_2000 = cleanedrs_data[cleanedrs_data['year_built'] >= 2000]['listPrice']

# Perform the t-test to compare average prices between the two groups
stat, p_value = ttest_ind(before_2000, after_2000)

#Results
print("T-test for Year Built and Listing Price:")
print("Statistic:", stat)
print("P-value:", p_value)

T-test for Year Built and Listing Price:
Statistic: -6.864363213577033
P-value: 7.60168749569103e-12


The p-value of 7.602e-12 is well below 0.05, so we reject the null hypothesis. This indicates that properties built before 2000 have a statistically different average listing price compared to those built after 2000. This difference might reflect how property age affects value, with newer homes possibly commanding higher prices.

### Do Properties with Larger Lot Sizes (lot_sqft) Command Higher Listing Prices?

Hypotheses:

- Null Hypothesis (H0): There is no significant correlation between lot size and listing price.
- Alternate Hypothesis (H1): There is a significant correlation between lot size and listing price.

**Test:** We use a correlation test here because we want to examine the relationship between two continuous variables (lot size and listing price).

In [20]:
# Calculate the correlation between lot size and listing price
correlation = cleanedrs_data['lot_sqft'].corr(cleanedrs_data['listPrice'])

#Print the correlation value
print("Correlation between Lot Size and Listing Price:", correlation)


Correlation between Lot Size and Listing Price: 0.28764119363384766


The correlation coefficient of 0.288 suggests a weak positive relationship between lot size and listing price. This means that while larger lot sizes tend to have higher listing prices, the effect is modest, and other factors likely play a stronger role in determining price.

### Does Having a Garage Significantly Influence the Listing Price?

Hypotheses:

- Null Hypothesis (H0): The average listing price is not significantly different between properties with and without a garage.
- Alternate Hypothesis (H1): The average listing price is significantly different between properties with and without a garage.

**Test:** Since we’re comparing two groups (properties with garages and without), a t-test is appropriate.


In [21]:
# Separate data into properties with garages and without garages
garage = cleanedrs_data[cleanedrs_data['garage'] == 1]['listPrice']
no_garage = cleanedrs_data[cleanedrs_data['garage'] == 0]['listPrice']

# Perform the t-test to compare average prices between properties with and without a garage
stat, p_value = ttest_ind(garage, no_garage)

#Results
print("T-test for Garage Presence and Listing Price:")
print("Statistic:", stat)
print("P-value:", p_value)


T-test for Garage Presence and Listing Price:
Statistic: 0.43911387665331897
P-value: 0.6609644734613036


Since the p-value (0.661) is greater than 0.05, we fail to reject the null hypothesis. This result indicates that there is no statistically significant difference in listing prices between properties with and without garages. In practical terms, having a garage does not appear to affect a property’s listing price, suggesting that buyers may not place much value on a garage when determining what they’re willing to pay.

### Is there a significant difference in prices across different property types?

Hypotheses:

- Null Hypothesis (H0): The average listing price does not significantly differ between different property types.
- Alternate Hypothesis (H1): The average listing price differs significantly between different property types.

**Test**: Since we're comparing mean prices across multiple groups an ANOVA test is the most appropriate.


In [22]:
# Group prices by property type
groups = [group["listPrice"].values for name, group in cleanedrs_data.groupby("type")]

# Perform the ANOVA test to see if property type impacts price
stat, p_value = f_oneway(*groups)

# Print the test statistic and p-value
print("ANOVA Test for Property Type and Listing Price:")
print("Statistic:", stat)
print("P-value:", p_value)


ANOVA Test for Property Type and Listing Price:
Statistic: 28.019234787925548
P-value: 1.3104889924714247e-52


With a very low p-value (1.310e-52), far below 0.05, we reject the null hypothesis. This result suggests a statistically significant difference in listing prices across different property types. This means that property types, such as single-family homes, condos, and townhouses, do have distinct average prices, likely due to different features, sizes, or amenities that appeal to buyers differently.

## Non-Technical Summary of Findings

1. Bedroom Count: The number of bedrooms significantly impacts property prices. Homes with more bedrooms tend to have higher prices, likely because they offer more space, which is generally more attractive to buyers.

2. Year Built: There is a notable price difference between properties built before 2000 and those built after. Newer homes seem to have higher values, possibly because they require less maintenance or have modern features that buyers want.

3. Lot Size: Although larger lots are somewhat more expensive, the impact of lot size on price is relatively weak. This suggests that factors other than lot size play a more important role in setting property prices.

4. Garage Presence: Surprisingly, having a garage does not significantly affect listing price. This means buyers are generally not willing to pay more just because a property has a garage.

5. Property Type: Different types of properties, like single-family homes, condos, and townhouses, have significantly different average prices. This reflects that buyers value these property types differently, possibly due to their unique features or the lifestyles they support.

These findings provide insights into what truly affects property values. While factors like the number of bedrooms, property age, and type have strong impacts on listing prices, others, like lot size and garage presence, seem less important. This information can be valuable for buyers, sellers, and real estate professionals in making informed decisions about property values and investments.